locked-asset/simple-lock-energy/src/unlock_with_penalty.rs

```diff
@@ -197,25 +208,90 @@ pub trait UnlockWithPenaltyModule:
+    fn burn_penalty(&self, token_id: TokenIdentifier, token_nonce: Nonce, fees_amount: &BigUint) {
         let fees_burn_percentage = self.fees_burn_percentage().get();
         let burn_amount = fees_amount * fees_burn_percentage as u64 / MAX_PERCENTAGE as u64;
         let remaining_amount = fees_amount - &burn_amount;
 
         if burn_amount > 0 {
-            self.send().esdt_local_burn(&token_id, 0, &burn_amount);
+            self.send()
+                .esdt_local_burn(&token_id, token_nonce, &burn_amount);
         }
         if remaining_amount > 0 {
-            self.send_fees_to_collector(token_id, remaining_amount);
+            if self.fees_from_penalty_unlocking().is_empty() {
+                // First fee deposit of the week
+                self.fees_from_penalty_unlocking()
+                    .set(NonceAmountPair::new(token_nonce, remaining_amount));
+            } else {
+                self.merge_fees_from_penalty(token_nonce, remaining_amount)
+            }
         }
+
+        // Only once per week
+        self.send_fees_to_collector();
     }
 
-    fn send_fees_to_collector(&self, token_id: TokenIdentifier, amount: BigUint) {
-        let sc_address = self.fees_collector_address().get();
-        self.fees_collector_proxy_builder(sc_address)
-            .deposit_swap_fees()
-            .add_esdt_token_transfer(token_id, 0, amount)
-            .execute_on_dest_context_ignore_result();
+    /// Merges new fees with existing fees and saves in storage
+    fn merge_fees_from_penalty(&self, token_nonce: Nonce, new_fee_amount: BigUint) {
+        let locked_token_mapper = self.locked_token();
+        let existing_nonce_amount_pair = self.fees_from_penalty_unlocking().get();
+        let mut payments = PaymentsVec::new();
+        payments.push(EsdtTokenPayment::new(
+            locked_token_mapper.get_token_id(),
+            token_nonce,
+            new_fee_amount,
+        ));
+        payments.push(EsdtTokenPayment::new(
+            locked_token_mapper.get_token_id(),
+            existing_nonce_amount_pair.nonce,
+            existing_nonce_amount_pair.amount,
+        ));
+
+        let new_locked_amount_attributes = self.merge_tokens(payments, OptionalValue::None);
+
+        let sft_nonce = self.get_or_create_nonce_for_attributes(
+            &locked_token_mapper,
+            &new_locked_amount_attributes
+                .attributes
+                .original_token_id
+                .clone()
+                .into_name(),
+            &new_locked_amount_attributes.attributes,
+        );
+
+        let new_locked_tokens = locked_token_mapper
+            .nft_add_quantity(sft_nonce, new_locked_amount_attributes.token_amount.clone());
+
+        self.fees_from_penalty_unlocking().set(NonceAmountPair::new(
+            new_locked_tokens.token_nonce,
+            new_locked_tokens.amount,
+        ));
+    }
+
+    #[payable("*")]
```

Comment:
this should not be payable - you do not transfer anything to this function.

Reply:
Correct, removed.